app/UpscanProxyModule.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import play.api.inject.{Binding, Module}
+import play.api.{Configuration, Environment}
+import uk.gov.hmrc.upscanuploadproxy.UploadUriGenerator
+
+class UpscanProxyModule extends Module {
+
+  @SuppressWarnings(Array("org.wartremover.warts.TryPartial"))
+  override def bindings(environment: Environment, configuration: Configuration): Seq[Binding[_]] =
+    Seq(
+      bind[UploadUriGenerator].to(new UploadUriGenerator {
+        override def uri(bucketName: String): String = s"https://$bucketName.s3.amazonaws.com"
```

Comment:
Is there any value of wiring up a uri generator over just putting the url template in config?